Announce 2024 user survey on the blog homepage

InessaPawson opened the pull request:
Add an announcement about the 2024 user survey to the callout bar on the blog homepage based on the discussion in https://github.com/scikit-learn/blog/issues/186

otieno-juma committed:
DOC: updated the scikit-learn website

otieno-juma committed:
added new changes

InessaPawson committed:
Add an announcement about the 2024 user survey to the callout bar

lesteve commented:
Thanks for getting the ball rolling, not sure this is absolutely crucial but the banner is white on the dark background:
![image](https://github.com/user-attachments/assets/9c86fba1-77de-43d2-8c68-e240e5968006)

glemaitre commented:
I'll give a look at this PR.

I was wondering if it would not be better to use the feature `annoucements` from `pydata-sphinx-theme` as shown in https://github.com/scikit-learn/scikit-learn/pull/29750

This would give the following default visual in dark and light mode.

<img width="1462" alt="image" src="https://github.com/user-attachments/assets/7ea46a85-b903-4396-ad86-1eec1bb4e884">

<img width="1468" alt="image" src="https://github.com/user-attachments/assets/9d068552-fcc6-4d75-beaa-f36b7d026d3f">

I find that the integration is a bit better. If you think this is not enough visible, we can probably increase the font.

@InessaPawson What do you think?

betatim commented:
I like the idea of using the "built in" announcement option from the theme. Without modifying its CSS (aka increasing the font)

glemaitre commented:
@InessaPawson Feel free to modify the `conf.py` in this PR. I'll close my own PR since it was just to check the rendering.

InessaPawson committed:
Spell out the name of the project

Spell out the name of the project as per suggestion from the code review.

Co-authored-by: Loïc Estève <loic.esteve@ymail.com>

InessaPawson committed:
Remove the banner section from index.scss

InessaPawson committed:
Remove 'update-container' section from templates/index.html

InessaPawson committed:
Use the ‘announcement’ key from pydata-sphinx-theme

Use the ‘announcement’ key from pydata-sphinx-theme to announce the 2024 user survey

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lesteve reviewed (approved):
LGTM, thanks!

Note: this adds the announcement banner to all the pages, but I think this is completely fine. I am mentioning this because https://github.com/scikit-learn/blog/issues/186 seemed more about adding it to a few selected pages.

Here is the link to the [rendered doc](https://output.circle-artifacts.com/output/job/ef2b8c24-2966-4537-b298-3db04e99eac7/artifacts/0/doc/index.html) for the next reviewer.

glemaitre reviewed (approved):
LGTM. Thanks @InessaPawson
